### PR TITLE
Fixing eslint errors in packages. Updated .eslintrc.json to enable es6.

### PR DESCRIPTION
**Pull request by srprash**
*Issue #, if available:*

*Description of changes:*
This PR fixes the following errors which were reported by eslint:
- indentations
- missing semicolons
- undefined variables
- invalid use of named params
- removing unused variables
- " to '

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/.eslintrc.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
     "env": {
         "node": true,
-        "mocha": true
+        "mocha": true,
+        "es6": true
```
Comment (luluzhao) [inline packages/core/.eslintrc.json]: Just wondering what is the benefit about enable es6?

File: packages/core/lib/logger.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 var winston = require('winston');
-var format = require('date-fns/format')
+var format = require('date-fns/format');
```
Comment (luluzhao) [inline packages/core/lib/logger.js]: Even if there is no ";" presenting at the end of each line, the code will still compile without throwing an error before?

Comment (srprash) [reply packages/core/lib/logger.js]: Yes. Semicolon is optional in Javascript and the code does compile and runs with or without it. However, as a best practice, it is advised to put use semicolons.

File: packages/core/.eslintrc.json (hunk 1 above)
Comment (srprash) [reply packages/core/.eslintrc.json]: Setting `es6` to `true` enables all of the ES6 globals such as Promise, Map, Set, etc. If not enabled, the eslint linter will raise the 'Promise not defined' error.

**Pull request merged**